core/src/main/scala/com/ironcorelabs/recrypt/CoreApi.scala

```diff
@@ -67,8 +67,12 @@ class CoreApi(
     //Realistically this should never happen because the private key shouldn't contain a value that goes to the zero point.
     .getOrElse(throw new Exception("privateKey was invalid.")))
 
-  def augmentPrivateKey(p1: PrivateKey, p2: PrivateKey): PrivateKey =
-    PrivateKey(internal.Fp.bigIntToByteVector((privateKeyTransform(p1) + privateKeyTransform(p2)).toBigInt))
+  /**
+   * Augment the private key with another. This does addition in Fr instead of Fp so it matches up the cycle of the curve.
+   */
+  def augmentPrivateKey(p1: PrivateKey, p2: PrivateKey): PrivateKey = {
+    PrivateKey(internal.Fp.bigIntToByteVector(internal.positiveMod(privateKeyTransform(p1).toBigInt + privateKeyTransform(p2).toBigInt, internal.Fp.Order)))
```

Comment:
Seeing `Fp` in there makes me think it is not doing addition in `Fr`

Reply:
The math is done in BigInt, modded by Fr, then put back into a byte vector which is wrapped up in the private key type.

Reply:
Isn't the second argument to `positiveMod` the modulus? I figured that `internal.Fp.Order` should be `p`.

Reply:
Well, it isn't `p`, it is `r`. And will stay that way for the time being.